src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -306,7 +306,7 @@ private List<LibraryDependency> GetDependenciesFromSpecRestoreMetadata(PackageSp
                 // Note: Only add in dependencies that are in the filtered list to avoid getting the wrong TxM
                 dependencies.AddRange(childReferences
                     .Where(reference => filteredExternalDependencies.Contains(reference.ProjectName))
-                    .Select(reference => new LibraryDependency
+                    .Select(reference => new LibraryDependency(noWarn: Array.Empty<NuGetLogCode>())
                     {
                         LibraryRange = new LibraryRange
```

Comment:
Not directly related to this change, but how often do we end up parsing the same `VersionRange`? Can we have a singleton for `"1.0.0"`?

Reply:
Using NuGet.Insights to check all packages on nuget.org, of packages that have at least one dependency, almost exactly 1% of those dependencies are for version 1.0.0 (well, `[1.0.0, )`). That doesn't mean the same will be true for this code path (project references where the other project doesn't specify a version will end up with 1.0.0 from memory), but I think nuget.org dependencies is the only data we have.

Reply:
Thanks for getting that data.  Sounds like we'd want to pursue any such change in a different pull request anyway.